soc/xtensa/intel_adsp/cavs/gdbstub.c

```diff
@@ -0,0 +1,1394 @@
+		.idx = 0x12f,
+		.regno = 0x12f,
+		.byte_size = 4,
+		.gpkt_offset = 192,
+	},
+	{
+		/* AR48 */
+		.idx = 0x130,
+		.regno = 0x130,
+		.byte_size = 4,
+		.gpkt_offset = 196,
+	},
+	{
+		/* AR49 */
+		.idx = 0x131,
+		.regno = 0x131,
+		.byte_size = 4,
+		.gpkt_offset = 200,
+	},
+	{
+		/* AR50 */
+		.idx = 0x132,
+		.regno = 0x132,
+		.byte_size = 4,
+		.gpkt_offset = 204,
+	},
+	{
+		/* AR51 */
+		.idx = 0x133,
+		.regno = 0x133,
+		.byte_size = 4,
+		.gpkt_offset = 208,
+	},
+	{
+		/* AR52 */
+		.idx = 0x134,
+		.regno = 0x134,
+		.byte_size = 4,
+		.gpkt_offset = 212,
+	},
+	{
+		/* AR53 */
+		.idx = 0x135,
+		.regno = 0x135,
+		.byte_size = 4,
+		.gpkt_offset = 216,
+	},
+	{
+		/* AR54 */
+		.idx = 0x136,
+		.regno = 0x136,
+		.byte_size = 4,
+		.gpkt_offset = 220,
+	},
+	{
+		/* AR55 */
+		.idx = 0x137,
+		.regno = 0x137,
+		.byte_size = 4,
+		.gpkt_offset = 224,
+	},
+	{
+		/* AR56 */
+		.idx = 0x138,
+		.regno = 0x138,
+		.byte_size = 4,
+		.gpkt_offset = 228,
+	},
+	{
+		/* AR57 */
+		.idx = 0x139,
+		.regno = 0x139,
+		.byte_size = 4,
+		.gpkt_offset = 232,
+	},
+	{
+		/* AR58 */
+		.idx = 0x13a,
+		.regno = 0x13a,
+		.byte_size = 4,
+		.gpkt_offset = 236,
+	},
+	{
+		/* AR59 */
+		.idx = 0x13b,
+		.regno = 0x13b,
+		.byte_size = 4,
+		.gpkt_offset = 240,
+	},
+	{
+		/* AR60 */
+		.idx = 0x13c,
+		.regno = 0x13c,
+		.byte_size = 4,
+		.gpkt_offset = 244,
+	},
+	{
+		/* AR61 */
+		.idx = 0x13d,
+		.regno = 0x13d,
+		.byte_size = 4,
+		.gpkt_offset = 248,
+	},
+	{
+		/* AR62 */
+		.idx = 0x13e,
+		.regno = 0x13e,
+		.byte_size = 4,
+		.gpkt_offset = 252,
+	},
+	{
+		/* AR63 */
+		.idx = 0x13f,
+		.regno = 0x13f,
+		.byte_size = 4,
+		.gpkt_offset = 256,
+	},
+	{
+		/* LBEG */
+		.idx = 512,
+		.regno = 0x0200,
+		.byte_size = 4,
+		.gpkt_offset = 260,
+		.stack_offset = BSA_LBEG_OFF,
+	},
+	{
+		/* LEND */
+		.idx = 513,
+		.regno = 0x0201,
+		.byte_size = 4,
+		.gpkt_offset = 264,
+		.stack_offset = BSA_LEND_OFF,
+	},
+	{
+		/* LCOUNT */
+		.idx = 514,
+		.regno = 0x0202,
+		.byte_size = 4,
+		.gpkt_offset = 268,
+		.stack_offset = BSA_LCOUNT_OFF,
+	},
+	{
+		/* SAR */
+		.idx = 515,
+		.regno = 0x0203,
+		.byte_size = 4,
+		.gpkt_offset = 272,
+		.stack_offset = BSA_SAR_OFF,
+	},
+	{
+		/* BR */
+		.idx = 516,
+		.regno = 0x204,
+		.byte_size = 4,
+		.is_read_only = 1,
+	},
+	{
+		/* PREFCTL */
+		.idx = 552,
+		.regno = 0x0228,
+		.byte_size = 4,
+		.is_read_only = 1,
+	},
+	{
+		/* WINDOWBASE */
+		.idx = 584,
+		.regno = 0x0248,
+		.byte_size = 4,
+		.gpkt_offset = 280,
+		.is_read_only = 1,
+	},
+	{
+		/* WINDOWSTART */
+		.idx = 585,
+		.regno = 0x0249,
+		.byte_size = 4,
+		.gpkt_offset = 284,
+		.is_read_only = 1,
+	},
+	{
+		/* PS */
+		.idx = 742,
+		.regno = 0x02E6,
+		.byte_size = 4,
+		.gpkt_offset = 296,
+		.stack_offset = BSA_PS_OFF,
+	},
+	{
+		/* THREADPTR */
+		.idx = 999,
+		.regno = 0x03E7,
+		.byte_size = 4,
+		.gpkt_offset = 300,
+#ifdef CONFIG_THREAD_LOCAL_STORAGE
+		/* Only saved in stack if TLS is enabled */
```

Comment:
/* comment says the same thing that the ifdef did */

Also, stylewise you can use an IF_ENABLED() here to save a few lines for readability.